static/js/components/dashboard/courses/StatusMessages.js

```diff
@@ -311,15 +322,28 @@ export const calculateMessages = (props: CalculateMessagesProps) => {
   } else {
     if (hasFailedCourseRun(course) && !hasPassedCourseRun(course)) {
       const date = run => formatDate(run.course_start_date)
+      const msg = run => {
+        let enrollmentDateMessage = ""
+        if (
+          !R.isNil(run.enrollment_start_date) &&
+          !R.isEmpty(run.enrollment_start_date) &&
```

Comment:
are this check and the check on the line above are performed by `isEnrollableRun`?

Reply:
yes they do same, i just refactored

Reply:
reverted, here i am adding these checks to make sure that enrollment start date is available and then it is not a `Enrollable Run` so that i can create a string message

Reply:
oh ok I understand, carry on :)